Exiled.Events/Handlers/Scp3114.cs

```diff
@@ -44,33 +43,66 @@ public static class Scp3114
+
+        /// <summary>
+        ///     Invoked before a player is slapped.
+        /// </summary>
+        public static Event<SlappingPlayerEventArgs> SlappingPlayer { get; set; } = new();
+
+        /// <summary>
+        ///     Invoked before triggering the strangle cooldown.
+        /// </summary>
+        public static Event<StranglingFinishedEventArgs> StranglingFinished { get; set; } = new();
+
+        /// <summary>
+        ///     Called before disguising to a new Roles.
         /// </summary>
         /// <param name="ev">The <see cref="DisguisingEventArgs" /> instance.</param>
         public static void OnDisguising(DisguisingEventArgs ev) => Disguising.InvokeSafely(ev);
 
         /// <summary>
-        ///     Called before diguising to a new Roles.
+        ///     Called before disguising to a new Roles.
         /// </summary>
         /// <param name="ev">The <see cref="DisguisedEventArgs" /> instance.</param>
         public static void OnDisguised(DisguisedEventArgs ev) => Disguised.InvokeSafely(ev);
 
         /// <summary>
-        ///     Called before diguising to a new Roles.
+        ///     Called before disguising to a new Roles.
         /// </summary>
         /// <param name="ev">The <see cref="TryUseBodyEventArgs" /> instance.</param>
         public static void OnTryUseBody(TryUseBodyEventArgs ev) => TryUseBody.InvokeSafely(ev);
 
         /// <summary>
-        ///     Called before diguising to a new Roles.
+        ///     Called before disguising to a new Roles.
         /// </summary>
         /// <param name="ev">The <see cref="RevealedEventArgs" /> instance.</param>
         public static void OnRevealed(RevealedEventArgs ev) => Revealed.InvokeSafely(ev);
 
         /// <summary>
-        ///     Called before diguising to a new Roles.
+        ///     Called before disguising to a new Roles.
         /// </summary>
         /// <param name="ev">The <see cref="RevealingEventArgs" /> instance.</param>
         public static void OnRevealing(RevealingEventArgs ev) => Revealing.InvokeSafely(ev);
+
+        /// <summary>
+        ///     Called before strangling a player.
+        /// </summary>
+        /// <param name="ev">The <see cref="StranglingEventArgs"/> instance.</param>
+        public static void OnStrangling(StranglingEventArgs ev) => Strangling.InvokeSafely(ev);
+
+        /// <summary>
+        ///     Called after Scp3114 finishes strangling a player.
+        /// </summary>
+        /// <param name="ev">The <see cref="StranglingFinishedEventArgs" /> instance.</param>
+        public static void OnStranglingFinished(StranglingFinishedEventArgs ev) => StranglingFinished.InvokeSafely(ev);
+
+        /// <summary>
+        ///     Called before Scp3114 slaps a player.
+        /// </summary>
+        /// <param name="ev">The <see cref="SlappingPlayerEventArgs" /> instance.</param>
+        public static void OnSlappingPlayer(SlappingPlayerEventArgs ev) => SlappingPlayer.InvokeSafely(ev);
```

Comment:
i will Verify if that is really needed (or if it's can simply use HurtEvent)

Reply:
It doesnt contain the damage handler directly, but you could probably replace the method that contains it directly by creating a new instance. The event still gives you access to the same general info, but the importance is that you can change the hume reward which is why I feel that it has justification to be a separate event.

I think if we replace the method, a prefix is probably more fitting because we will have rewrote a majority of the method to the point that other transpilers and prefixes would probably break anyways.

Patched Method:
```cs
public override void DamagePlayers()
	{
		Transform playerCameraReference = base.Owner.PlayerCameraReference;
		ReferenceHub primaryTarget = DetectedPlayers.GetPrimaryTarget(playerCameraReference);
		if (!(primaryTarget == null))
		{
			DamagePlayer(primaryTarget, DamageAmount);
			if (HasAttackResultFlag(AttackResult.KilledHuman))
			{
				this.ServerOnKill?.Invoke();
			}
			if (HasAttackResultFlag(AttackResult.AttackedHuman))
			{
				this.ServerOnHit?.Invoke();
			}
			float a = _humeShield.CurValue + 25f;
			_humeShield.CurValue = Mathf.Min(a, _humeShield.MaxValue);
		}
	}
```

Player Damage Method:
```cs
public virtual void DamagePlayer(ReferenceHub hub, float damage)
{
	PlayerStats playerStats = hub.playerStats;
	if (playerStats.DealDamage(DamageHandler(damage)))
	{
		_syncAttack |= AttackResult.AttackedHuman;
		if (!(playerStats.GetModule<HealthStat>().CurValue > 0f))
		{
			_syncAttack |= AttackResult.KilledHuman;
		}
	}
}
```

Reply:
Also it allows cancelling the method before processing the damage which can be used for custom Friendly Fire Arguments